src/content/developers/docs/smart-contracts/composability/index.md

```diff
@@ -10,6 +10,53 @@ incomplete: true
+
+## How does composability work? {#how-does-composability-work}
+
+Ethereum smart contracts are like public APIs, so anyone can interact with the contract or integrate them into dapps for added functionality. Smart contract composability generally works off three principles: modularity, autonomy, and discoverability:  
+
+**1. Modularity**: This is the ability of individual components to perform a specific task. In Ethereum, every smart contract has specific use-case (as shown in the Uniswap example).
+
+**2. Autonomy**: Composable components must be able to operate independently. Each smart contract in Ethereum is self-executing and can function without relying on other parts of the system. 
+
+**3. Discoverability**: Developers cannot call external contracts or integrate software libraries into applications if the former are not publicly available. By design, smart contracts are open-source; anyone can call a smart contract or fork a codebase. 
+
+## Benefits of composability {#benefits-of-composability}
+
+### Shorter development cycle {#shorter-development-cycle}
+
+Composability reduces the work that developers have to do when creating [dapps](/dapps/#what-are-dapps). [As Naval Ravikant puts it:](https://twitter.com/naval/status/1444366754650656770) "Open source means every problem has to be solved once."
+
+If there is a smart contract that solves one problem, other developers can reuse it, so they don’t have to solve the same problem. This way, developers can take existing software libraries and add extra functionality to create new dapps.
+
+### Greater innovation {#greater-innovation}
+
+Composability encourages innovation and experimentation because developers are free to reuse, modify, duplicate, or integrate open-source code to create desired results. As a result, development teams spend less time on basic functionality and can allocate more time experimenting with new features.
+
+
+### Better user experience {#better-user-experience}
+
+Interoperability between components of the Ethereum ecosystem improves the user experience. Users can access greater functionality when dapps integrate external smart contracts than in a fragmented ecosystem where applications cannot communicate.
+
+## Examples of composability in Ethereum {#composability-in-ethereum}
+
+### Token swaps {#token-swaps}
+
+If you create a dapp that requires transactions to be paid in ETH, you can allow users to pay in other ERC-20 tokens by integrating token swap logic. The code will automatically convert the user’s token to ETH before the contract executes the called function.
+
+### Governance {#governance}
+
+Building bespoke governance systems for a [DAO](/dao/) can be expensive and time-consuming. Instead, you could use an open-source governance toolkit, such as [Aragon Client](https://client.aragon.org/), to bootstrap your DAO to quickly create a governance framework. 
```

Comment:
@minimalsm Just noting this product callout. Not necessarily against keeping it since it does demonstrate SC composability, but obviously we've also been discussing trying to not open the flood gates for listings. Curious if you have thoughts.